Model/Service/CreateInvoice.php

```diff
@@ -100,7 +100,7 @@ public function createInvoiceGeneralSetting(Order $order): bool
     {
         $this->payment = $order->getPayment();
 
-        $this->addTransactionData();
+        $this->addTransactionData($this->payment);
```

Comment:
if both method and property are on the same object why do you need to add it as a parameter here?

Reply:
I use the addTransactionData function on app/code/Buckaroo/Magento2/Observer/SetTransactionOnInvoiceObserver.php:94

This is why I changed the signature of the function.

Thanks. Now I can remove for now $this->payment.